agent/exec/controller_test.go

```diff
@@ -368,6 +368,25 @@ func TestShutdown(t *testing.T) {
 	})
 }
 
+func TestRemove(t *testing.T) {
```

Comment:
`TestDesiredStateRemove`

Include a comment explaining that we are testing that the agent maintains `SHUTDOWN` as the maximum state in the agent.